src/App.tsx

```diff
@@ -1,5 +1,88 @@
-import React, { FC } from "react";
+import { FC, useState, useEffect } from "react";
+import { Typography } from "@mui/material";
+import { ITask, Task } from "./Task";
+import { CreateTaskForm } from "./CreateTaskForm";
 
-const App: FC = () => <div>App</div>;
+const useTasks = () => {
+  const getInitialState = () => {
+    const tasks = localStorage.getItem("tasks");
+    if (tasks) {
+      return JSON.parse(tasks);
+    }
+    return [];
+  };
+
+  const [tasks, setTasks] = useState<ITask[]>(getInitialState());
+
+  useEffect(() => {
+    localStorage.setItem("tasks", JSON.stringify(tasks));
+  }, [tasks]);
+
+  return {
+    tasks,
+    setTasks,
+  };
+};
+
+const App: FC = () => {
+  const { tasks, setTasks } = useTasks();
+
+  const onNewTaskCreate = (
+    { name, description }: Omit<ITask, "id">,
+    { resetForm }: { resetForm: Function }
+  ) => {
+    setTasks((tasks) => [
+      ...tasks,
+      {
+        id: crypto.randomUUID(),
+        name,
+        description,
+      },
+    ]);
+    resetForm();
+  };
+
+  const onTaskDelete = (id: ITask["id"]) => () => {
+    setTasks((tasks) => tasks.filter((task) => task.id !== id));
+  };
+
+  const onTaskComplete = (id: ITask["id"]) => () => {
+    setTasks((tasks) =>
+      tasks.map((task) => {
+        if (task.id !== id) return task;
+        return {
+          ...task,
+          isComplete: true,
+        };
+      })
+    );
+  };
+
+  return (
+    <div className="layout">
+      <header className="header">
+        <Typography mb={2} variant="h3" component="h1">
+          ToDo List
+        </Typography>
+      </header>
+      <div className="aside">
+        <CreateTaskForm onNewTaskCreate={onNewTaskCreate} />
```

Comment:
Great, you set props with function, which adding to state. Often people send setState func in props